src/modules/admin/components/services/ServiceTypeDialog.tsx

```diff
@@ -0,0 +1,197 @@
+import { isPickListOption } from '@/modules/form/types';
+import { AdminDashboardRoutes } from '@/routes/routes';
+import {
+  CreateServiceTypeMutation,
+  PickListOption,
+  PickListType,
+  ServiceTypeFieldsFragment,
+  UpdateServiceTypeMutation,
+  useCreateServiceTypeMutation,
+  useGetPickListQuery,
+  useUpdateServiceTypeMutation,
+  ValidationError,
+} from '@/types/gqlTypes';
+import { evictQuery } from '@/utils/cacheUtil';
+import { generateSafePath } from '@/utils/pathEncoding';
+
+interface ServiceTypeDialogProps {
+  serviceType?: ServiceTypeFieldsFragment;
+  dialogOpen: boolean;
+  closeDialog: () => void;
+}
+
+const ServiceTypeDialog: React.FC<ServiceTypeDialogProps> = ({
+  serviceType,
+  dialogOpen,
+  closeDialog,
+}) => {
+  const navigate = useNavigate();
+
+  const [name, setName] = useState(serviceType?.name || '');
+  const [supportsBulkAssignment, setSupportsBulkAssignment] = useState(
+    serviceType?.supportsBulkAssignment || false
+  );
+  const [serviceCategory, setServiceCategory] = useState<PickListOption | null>(
+    null
+  );
+  const [errors, setErrors] = useState<ErrorState>(emptyErrorState);
+
+  const mutationInput = useMemo(() => {
+    return {
+      name,
+      supportsBulkAssignment,
+      // Relies on custom elements in CreatableFormSelect lacking `label` attr
+      serviceCategoryId: !!serviceCategory?.label ? serviceCategory.code : null,
+      serviceCategoryName: !serviceCategory?.label
+        ? serviceCategory?.code
+        : null,
+    };
+  }, [
+    name,
+    supportsBulkAssignment,
+    serviceCategory?.label,
+    serviceCategory?.code,
+  ]);
+
+  const {
+    data: { pickList } = {},
+    loading: pickListLoading,
+    error: pickListError,
+  } = useGetPickListQuery({
+    variables: { pickListType: PickListType.CustomServiceCategories },
+  });
+
+  useEffect(() => {
+    // When serviceType exists (meaning this is an edit dialog, not a new unsaved record),
+    // populate its existing serviceCategory into the Service Category dropdown.
+    // Do this in a useEffect so that it works regardless of whether the pickList is cached or fetched from network
+    if (serviceType && pickList) {
+      const pickListOption = pickList.find(
+        (p) => p.code === serviceType.serviceCategory.id
+      );
+      setServiceCategory(pickListOption || null);
+    }
+  }, [pickList, serviceType, dialogOpen]);
+
+  const onClose = useCallback(() => {
+    closeDialog();
+    // Don't reset to defaults here - that would cause the update dialog to show null values on reopen
+    setErrors(emptyErrorState);
+  }, [setErrors, closeDialog]);
+
+  const onMutationCompleted = useCallback(
+    (errors: ValidationError[] = []) => {
+      if (errors.length > 0) {
+        setErrors(partitionValidations(errors));
+      } else {
+        evictQuery('serviceTypes');
+        evictQuery('pickList', {
+          pickListType: PickListType.CustomServiceCategories,
+        });
```

Comment:
I think it's best to avoid manually manipulating the cache when possible, and instead tell the apollo query when to re-fetch. The table already uses cache-and-network by default, so the refetch will happen anyway when you navigate back to the table, it just takes a moment.

For the picklist, you can remove this eviction and add `fetchPolicy: 'cache-and-network',` to the `useGetPickListQuery` query. 

https://www.apollographql.com/docs/react/data/queries#supported-fetch-policies